From 5f41055235786657509233557a3ca2950c401ec5 Mon Sep 17 00:00:00 2001
From: marcin <marcin@ipv8.pl>
Date: Wed, 15 Jun 2022 03:46:13 +0200
Subject: [PATCH] sound/rockchip/i2s: Don't disable mclk on suspend

This is a workaround to fix an issue with high-pitch sound after
suspend.

This patch is actually authored by Biktorgj
---
 sound/soc/rockchip/rockchip_i2s.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/rockchip/rockchip_i2s.c b/sound/soc/rockchip/rockchip_i2s.c
index 4ce5d2579387..1f9d4e5e36b2 100644
--- a/sound/soc/rockchip/rockchip_i2s.c
+++ b/sound/soc/rockchip/rockchip_i2s.c
@@ -61,7 +61,7 @@ static int i2s_runtime_suspend(struct device *dev)
 	struct rk_i2s_dev *i2s = dev_get_drvdata(dev);
 
 	regcache_cache_only(i2s->regmap, true);
-	clk_disable_unprepare(i2s->mclk);
+	//clk_disable_unprepare(i2s->mclk);
 
 	return 0;
 }
-- 
GitLab

